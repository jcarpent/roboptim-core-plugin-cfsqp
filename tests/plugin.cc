// Copyright (C) 2009 by Thomas Moulard, AIST, CNRS, INRIA.
//
// This file is part of the roboptim.
//
// roboptim is free software: you can redistribute it and/or modify
// it under the terms of the GNU Lesser General Public License as published by
// the Free Software Foundation, either version 3 of the License, or
// (at your option) any later version.
//
// roboptim is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU Lesser General Public License for more details.
//
// You should have received a copy of the GNU Lesser General Public License
// along with roboptim.  If not, see <http://www.gnu.org/licenses/>.

#include <iostream>
#include <boost/mpl/vector.hpp>
#include <boost/numeric/ublas/io.hpp>
#include <boost/variant/get.hpp>

#include <roboptim/core/solver-factory.hh>

#include "common.hh"
#include "hs071.hh"

using namespace roboptim;
typedef boost::mpl::vector<LinearFunction, DerivableFunction> clist_t;
typedef Solver<DerivableFunction, clist_t> solver_t;

int run_test ()
{
  F f;

  solver_t::problem_t pb (f);
  initialize_problem<solver_t::problem_t,
    roboptim::DerivableFunction> (pb);

  // Initialize solver
  SolverFactory<solver_t> factory ("cfsqp", pb);
  solver_t& solver = factory ();

  // Compute the minimum and retrieve the result.
  solver_t::result_t res = solver.minimum ();

  // Display solver information.
  std::cout << solver << std::endl;

  // Display the result.
  std::cout << res << std::endl;
  return 0;
}


GENERATE_TEST ()
