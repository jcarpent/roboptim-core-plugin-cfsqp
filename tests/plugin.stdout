Solver:
  Problem:
    a * d * (a + b + c) + d (twice derivable function)
    Argument's bounds: (1, 5), (1, 5), (1, 5), (1, 5)
    Argument's scales: 1, 1, 1, 1
    Number of constraints: 2
    Constraint 0
        a * b * c * d (twice derivable function)
        Bounds: (25, inf)
        Scales: 1
        Initial value: [1](25)
        
    Constraint 1
        a * a + b * b + c * c + d * d (twice derivable function)
        Bounds: (40, 40)
        Scales: 1
        Initial value: [1](52) (constraint not satisfied)
        
    Starting point: [4](1,5,5,1)
    Starting value: [1](12)
    Infinity value (for all functions): inf
  Parameters:
    cfsqp.bigbnd (symbolizes infinity): 1e+10
    cfsqp.eps (final norm requirement for the Newton direction): 1e-08
    cfsqp.epseqn (maximum violation of nonlinear equality constraint): 1e-08
    cfsqp.iprint (logging level): 0
    cfsqp.mode (CFSQP mode): 100
    cfsqp.udelta (perturbation size used in CFSQP finite differences algorithm): 1e-08
    max-iterations (number of iterations): 50
    
CFSQP specific variables: 
  Nineq: 1
  Nineqn: 1
  Neq: 1
  Neqn: 1
  CFSQP constraints: (0, 1), (1, 1)
Result: 
    Size (input, output): 4, 1
    X: [4](1,4.743,3.82115,1.37941)
    Value: [1](14.5723)
    Constraints values: [2](25,40)
    Lambda: [2](0,0)
    Warnings: New iterate equivalent to previous one.
